Python/Product/PythonTools/PythonTools/LanguageServerClient/PythonLanguageClient.cs

```diff
@@ -367,12 +367,17 @@ private async Task NotifyWithParametersAsync(string request, object parameters)
         }
 
         private LanguageServerSettings.PythonSettings GetSettings(Uri scopeUri = null) {
+            // on shutdown return null
+            if (_clientContexts.Count() == 0) {
+                return null;
+            }
             IPythonLanguageClientContext context = null;
             if (scopeUri == null) {
-                // REPL context has null RootPath
+                // REPL context has null RootPath 
                 context = _clientContexts.Find(c => c.RootPath == null);
                 if (context == null) {
-                    return null;
+                    // use first clientcontext as default
+                    context = _clientContexts.First();
```

Comment:
Is there a FirstOrNull thing? I wasn't sure if this would throw if there's no items in the list.

Reply:
Oh it's called 'FirstOrDefault'. Maybe you should use that?

Reply:
i check count at the top of the function. but ya. that would be safer